File: Modules/HumanResource/OrganisationHierarchy/Occupation/OccupationForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.OrganisationHierarchy.Forms;

[FormScript("OrganisationHierarchy.Occupation")]
[BasedOnRow(typeof(OccupationRow), CheckNames = true)]
public class OccupationForm
{
    public string Name { get; set; }

    public string Description { get; set; }
}

File: Modules/ServerTypes/OrganisationHierarchy/OccupationForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface OccupationForm {
    Name: StringEditor;
    Description: StringEditor;
}

export class OccupationForm extends PrefixedContext {
    static formKey = 'OrganisationHierarchy.Occupation';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!OccupationForm.init)  {
            OccupationForm.init = true;

            var w0 = StringEditor;

            initFormType(OccupationForm, [
                'Name', w0,
                'Description', w0
            ]);
        }
    }
}